primitives/src/merkle_tree/namespaced_merkle_tree/mod.rs

```diff
@@ -93,7 +95,16 @@ where
 
 /// Trait indiciating that a struct can act as an orderable namespace
 pub trait Namespace:
-    Debug + Clone + CanonicalDeserialize + CanonicalSerialize + Default + Copy + Hash + Ord
+    Debug
+    + Clone
+    + CanonicalDeserialize
+    + CanonicalSerialize
+    + Default
+    + Copy
+    + Hash
+    + Ord
+    + Serialize
+    + for<'a> Deserialize<'a>
```

Comment:
You should use `DeserializeOwned` instead (`serde::de::DeserializeOwned`). I believe it's just an alias for this bound but serde has the right to change it in the future